src/common/queryParameterHooks.js

```diff
@@ -14,14 +14,13 @@ export const useReplaceQueryParameter = () => {
 
   return parameters => {
     parameters.forEach(({ key, value }) => {
-      if (value === undefined) {
-        searchParams.delete(key);
+      if (value === "") {
+        searchParams.delete(key); 
       } else {
         searchParams.set(key, value);
       }
     });
 
-    history.push(`${location.pathname}?${searchParams.toString()}`);
+    history.replace(`${location.pathname}?${searchParams.toString()}`);
```

Comment:
If you add a condition to render clear button only when the query exists you won't have to change history.push to history.replace to eliminate warning in the console. 😁